tools/profiles/develop.json

```diff
@@ -5,7 +5,7 @@
                    "-fmessage-length=0", "-fno-exceptions", "-fno-builtin",
                    "-ffunction-sections", "-fdata-sections", "-funsigned-char",
                    "-MMD", "-fno-delete-null-pointer-checks",
-                   "-fomit-frame-pointer", "-Os", "-g1"],
+                   "-fomit-frame-pointer", "-Os", "-g1", "-DMBED_TRAP_ERRORS_ENABLED=1"],
```

Comment:
Don't know how I feel about this change, but even if agreed it's definitely not a "fix" any more, so would need to be in a separate PR if you want the `debug`->`mbed_error_printf` change in a 5.11 patch. This would trigger a bunch of fatal errors for people who haven't done debug builds before.

Reply:
Will split the PR's